### PR TITLE
feat(sdk-core, sdk-coin-eth): add function to transfer nfts

### PR DESCRIPTION
**Pull request by mohammadalfaiyazbitgo**
Ticket: WP-1094

TICKET: WP-1094

### CODE REVIEW

**Comment by mohammadalfaiyazbitgo** [comment]
Transferred NFT with method here: https://goerli.etherscan.io/tx/0xba821690529fd637d368e7fbeff165dc7af60c177ca5a46fc7a984e0320c3304

File: modules/abstract-eth/src/abstractEthLikeNewCoins.ts [hunk 1]
```diff
@@ -2474,4 +2477,38 @@ export abstract class AbstractEthLikeNewCoins extends AbstractEthLikeCoin {
     }
     return Buffer.concat(parts);
   }
+
+  /**
+   * Build the data to transfer an ERC-721 or ERC-1155 token to another address
+   * @param params
+   */
+  buildNftTransferData(params: NFTTransferOptions & { fromAddress: string }): string {
```
Comment (margueriteblair) [inline modules/abstract-eth/src/abstractEthLikeNewCoins.ts:2485]: I see that this is only implemented for `abstractEthLikeNewCoins` and not `abstractEthLike`. That's intentional right? I'm not super familiar on the ETH set up, just wanna make sure that we're only expecting it to be for these new coins

File: modules/sdk-core/src/bitgo/wallet/wallet.ts [hunk 2]
```diff
@@ -2099,6 +2102,79 @@ export class Wallet implements IWallet {
...
+          }
+        }
+
+        const data = this.baseCoin.buildNftTransferData({ ...sendNftOptions, fromAddress: baseAddress });
+        return this.sendMany({
+          ...sendOptions,
+          recipients: [
+            {
+              address: sendNftOptions.tokenContractAddress,
+              amount: '0',
```
Comment (margueriteblair) [inline modules/sdk-core/src/bitgo/wallet/wallet.ts:2169]: Is there a case where we'd have any fees and it doesnt make sense for `amount` to be 0?

File: modules/abstract-eth/src/abstractEthLikeNewCoins.ts (hunk 1 above)
Comment (mohammadalfaiyazbitgo) [reply modules/abstract-eth/src/abstractEthLikeNewCoins.ts:2485]: I believe so, both `eth` & `polygon` extend from `abstractEthLikeNewCoins` which are the only chains we need to implement nfts for atm.

File: modules/sdk-core/src/bitgo/wallet/wallet.ts (hunk 2 above)
Comment (mohammadalfaiyazbitgo) [reply modules/sdk-core/src/bitgo/wallet/wallet.ts:2169]: Don't think so, the documentation https://developers.bitgo.com/guides/transact/nfts says to always set the amount to 0.

**Review by rlutzbitgo — APPROVED** [review]

File: modules/sdk-core/src/bitgo/wallet/wallet.ts [hunk 3]
```diff
@@ -2099,6 +2102,91 @@ export class Wallet implements IWallet {
...
+              data: data,
+            },
+          ],
+        });
```
Comment (alebusse) [inline modules/sdk-core/src/bitgo/wallet/wallet.ts:2156]: seems duplicated, a nit for another pr

**Review by alebusse — APPROVED** [review]

**Pull request merged**